File: app/src/main/java/in/ceeq/alec/provider/contentprovider/ContactDatabaseHelper.java
package in.ceeq.alec.provider.contentprovider;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

public class ContactDatabaseHelper extends SQLiteOpenHelper {

    private static final String DB_NAME = "contacts";

    private static final int DB_VERSION = 1;

    public ContactDatabaseHelper(Context context) {
        super(context, DB_NAME, null, DB_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {

        db.execSQL(ContactContract.Contacts.create());
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL(ContactContract.Contacts.drop());
        onCreate(db);
    }
}

File: app/src/main/java/in/ceeq/lyte/data/contentprovider/ContactDatabaseHelper.kt
package `in`.ceeq.lyte.data.contentprovider

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class ContactDatabaseHelper(context: Context) : SQLiteOpenHelper(context, DB_NAME, null, DB_VERSION) {

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(ContactContract.Contacts.create())
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL(ContactContract.Contacts.drop())
        onCreate(db)
    }

    companion object {
        private val DB_NAME = "contacts"
        private val DB_VERSION = 1
    }
}
